pg-extend/src/pg_datum.rs

```diff
@@ -370,6 +376,71 @@ where
     }
 }
 
+impl<'s, T> TryFromPgDatum<'s> for &[T]
+where
+    T: 's + TryFromPgDatum<'s>,
```

Comment:
So if we introduced the `PgCastable` trait and bound T with that, does it help anything?